src/controllers/groupe.controller.ts

```diff
@@ -0,0 +1,83 @@
+import { NextFunction, Request, Response } from 'express';
+import { Group } from '@prisma/client';
+import { CreateGroupDto } from '@dtos/group.dto';
+import groupService from '@services/groups.service';
+
+class GroupController {
+  public groupService = new groupService();
+
+  public getGroups = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const findAllGroupsData: Group[] = await this.groupService.findAllGroup();
+
+      res.status(200).json({ data: findAllGroupsData, message: 'findAll' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public getGroupById = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const groupId = Number(req.params.id);
+      const findOneGroupData: Group = await this.groupService.findGroupById(groupId);
+
+      res.status(200).json({ data: findOneGroupData, message: 'findOne' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public createGroup = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const groupData: CreateGroupDto = req.body;
+      const createGroupData: Group = await this.groupService.createGroup(groupData);
+
+      res.status(201).json({ data: createGroupData, message: 'created' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public updateGroup = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const groupId = Number(req.params.id);
+      const groupData: CreateGroupDto = req.body;
+      const updateGroupData: Group = await this.groupService.updateGroup(groupId, groupData);
+
+      res.status(200).json({ data: updateGroupData, message: 'updated' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public deleteGroup = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    try {
+      const groupId = Number(req.params.id);
+      const deleteGroupData: Group = await this.groupService.deleteGroup(groupId);
+
+      res.status(200).json({ data: deleteGroupData, message: 'deleted' });
+    } catch (error) {
+      next(error);
+    }
+  };
+
+  public joinGroup = async (req, res, next) => {
+    const groupId = Number(req.params.id);
+    const userId = req.body.userId;
+
+    const users = await this.groupService.joinGroup(userId, groupId);
+
+    res.status(200).json(users)
+  }
+
+  public leaveGroup = async (req, res, next) => {
+    const groupId = Number(req.params.id);
+    const userId = req.body.userId;
+
+    const users = await this.groupService.leaveGroup(userId, groupId);
+
+    res.status(200).json(users)
+  }
+}
+
+export default GroupController;
```

Comment:
1. Typing: The req, res, and next parameters in the joinGroup and leaveGroup methods are not typed. Try to always specify the types for better type checking and autocompletion features:
```
public joinGroup = async (req: Request, res: Response, next: NextFunction): Promise<void> => { /* ... */ }
public leaveGroup = async (req: Request, res: Response, next: NextFunction): Promise<void> => { /* ... */ }
```
2. Error handling: Each method is catching errors and passing them to the next function but there's no information about what next does with these errors. Ensure that there's a general error handling middleware in your Express app to properly handle these errors.
3. Dependency Injection: Instead of instantiating the groupService inside the GroupController, consider using dependency injection. This makes your code easier to test and maintain:
```
public groupService: GroupService;

constructor(groupService: GroupService) {
  this.groupService = groupService;
}
```
4. Route validation: There is no validation of incoming requests. Consider using a middleware for route validation, like express-validator or joi.
5. HTTP Status Codes: It is better to use HTTP Status Codes constants for making it more readable. Example:
```
res.status(HttpStatus.OK).json({ data: findAllGroupsData, message: 'findAll' });
```
6. Async/Await: You're not handling possible exceptions of async/await properly, use try-catch for async/await.
7. Response Format: The responses from different endpoints should have a consistent structure. Currently, joinGroup and leaveGroup are returning just a users array. It would be better to return it in the same format with data and message fields, like other endpoints.

Also DRY.
This pattern can be abstracted into a higher-order function. Here's how you can add this:
```
private handleRequest = (serviceFunction: (...args: any[]) => Promise<any>) => {
  return async (req: Request, res: Response, next: NextFunction): Promise<void> => {
    try {
      const result = await serviceFunction.call(this.groupService, ...Object.values(req.params), req.body);
      res.status(HttpStatus.OK).json({ data: result, message: 'success' });
    } catch (error) {
      next(error);
    }
  };
};

public getGroups = this.handleRequest(this.groupService.findAllGroup);

public getGroupById = this.handleRequest(this.groupService.findGroupById);

public createGroup = this.handleRequest(this.groupService.createGroup);

public updateGroup = this.handleRequest(this.groupService.updateGroup);

public deleteGroup = this.handleRequest(this.groupService.deleteGroup);

public joinGroup = this.handleRequest(this.groupService.joinGroup);

public leaveGroup = this.handleRequest(this.groupService.leaveGroup);
```